### PR TITLE
SynopticRA1 Internal Navigation

### PR DESCRIPTION
**Pull request by Patowhiz**
@africanmathsinitiative/developers this is ready for review
Partly Fixes issue  #72 and  #83

### CODE REVIEW

File: ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb [hunk 1]
```diff
@@ -41,7 +41,6 @@ Public Class ucrSynopticRA1
                     lstFields.Add(strFlagFieldName & ctrVFP.Tag)
                     AddHandler ctrVFP.ucrValue.evtValueChanged, AddressOf InnerControlValueChanged
                     AddHandler ctrVFP.ucrFlag.evtValueChanged, AddressOf InnerControlValueChanged
-                    AddHandler ctrVFP.evtGoToNextVFPControl, AddressOf GoToNextVFPControl
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb]: Why did you remove this line from here? How do we handle the event to go to the ext control?

Comment (Patowhiz) [reply ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb]: At the form level. I found it very easy to let the form handle the change of focus. It's the container of all the controls and is in the position to correctly determine what the next control is.
And also this prepares the Value Flag Period into not needing to know if its being used in a group or a single control. Which promotes abstraction and encapsulation of functionilties

Comment (Patowhiz) [reply ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb]: And that also helps in partly removing the necessity of tagging and also trying to check whether a control is enabled or disabled

Comment (maxwellfundi) [reply ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb]: before we change how this works, then we need to see if @dannyparsons  is happy with this change and how it would affect other things from how this was conceptualized.

Comment (Patowhiz) [reply ClimsoftVer4/ClimsoftVer4/ucrSynopticRA1.vb]: @dannyparsons this is how I'm handling change of focus on enter

```
 ''' <summary>
    ''' sets key down listeners for the form controls
    ''' </summary>
    Private Sub SetControlsKeyDownListners()
        Dim synopticControl As ucrSynopticRA1

        For Each formCtr As Control In Me.Controls
            If TypeOf formCtr Is ucrBaseDataLink Then
                If TypeOf formCtr Is ucrSynopticRA1 Then
                    'for ucrSynopticRA1 controls
                    synopticControl = DirectCast(formCtr, ucrSynopticRA1)
                    For Each synopCtr As Control In synopticControl.Controls
                        If TypeOf synopCtr Is ucrValueFlagPeriod Then
                            AddHandler DirectCast(synopCtr, ucrValueFlagPeriod).evtKeyDown, AddressOf GoToNextControl
                        End If
                    Next
                Else
                    'for the other base controls e.g station, year, month, day, hour slectors
                    AddHandler DirectCast(formCtr, ucrBaseDataLink).evtKeyDown, AddressOf GoToNextControl
                End If
            End If
        Next
    End Sub

    ''' <summary>
    ''' determines the next control to get focus on enter key  
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub GoToNextControl(sender As Object, e As KeyEventArgs)
        If e.KeyCode = Keys.Enter Then
            Dim bGoToNextControl As Boolean = False

            'do validations to determine whether to go to next control
            If TypeOf sender Is ucrValueFlagPeriod Then
                If DirectCast(sender, ucrValueFlagPeriod).PreValidateValue() Then
                    bGoToNextControl = True
                End If
            ElseIf TypeOf sender Is ucrBaseDataLink Then
                'for the other base controls e.g station, year, month, day, hour slectors
                If DirectCast(sender, ucrBaseDataLink).ValidateValue() Then
                    bGoToNextControl = True
                End If
            End If

            If bGoToNextControl Then
                Me.SelectNextControl(sender, True, True, True, True)
            End If

            'to handle the "noise"
            e.SuppressKeyPress = True
        End If
    End Sub
```
